lua_cjson.c

```diff
@@ -698,8 +700,21 @@ static void json_append_data(lua_State *l, json_config_t *cfg,
         len = lua_array_length(l, cfg, json);
         if (len > 0 || (len == 0 && !cfg->encode_empty_table_as_object))
             json_append_array(l, cfg, current_depth, json, len);
-        else
-            json_append_object(l, cfg, current_depth, json);
+        else {
+            int as_array = 0;
+            if (lua_getmetatable(l, -1)) {
+                lua_pushlightuserdata(l, &json_empty_array);
+                lua_gettable(l, LUA_REGISTRYINDEX);
```

Comment:
Shall we use `lua_rawget` here instead of `lua_gettable`? The former should be a little bit faster and we don't need any metamethods here anyway.